frontend/pages/hosts/details/HostDetailsPage/modals/SelectQueryModal/SelectQueryModal.tsx

```diff
@@ -1,54 +1,108 @@
 import React, { useState, useCallback, useContext } from "react";
-
+import { useQuery } from "react-query";
 import { filter, includes } from "lodash";
+import { InjectedRouter } from "react-router";
+
+import { TAGGED_TEMPLATES } from "utilities/helpers";
+
+import PATHS from "router/paths";
+
+import permissions from "utilities/permissions";
+
 import { AppContext } from "context/app";
+import { QueryContext } from "context/query";
+
+import queryAPI from "services/entities/queries";
 
-import Button from "components/buttons/Button";
-import Modal from "components/Modal";
 // @ts-ignore
 import InputFieldWithIcon from "components/forms/fields/InputFieldWithIcon";
-
+import Button from "components/buttons/Button";
+import Modal from "components/Modal";
 import DataError from "components/DataError";
-import permissions from "utilities/permissions";
-import { ISchedulableQuery } from "interfaces/schedulable_query";
+
+import {
+  IListQueriesResponse,
+  IQueryKeyQueriesLoadAll,
+  ISchedulableQuery,
+} from "interfaces/schedulable_query";
+import { API_ALL_TEAMS_ID } from "interfaces/team";
+import { DEFAULT_TARGETS_BY_TYPE } from "interfaces/target";
 
 export interface ISelectQueryModalProps {
   onCancel: () => void;
-  onQueryHostCustom: () => void;
-  onQueryHostSaved: (selectedQuery: ISchedulableQuery) => void;
-  queries: ISchedulableQuery[] | [];
-  queryErrors: Error | null;
   isOnlyObserver?: boolean;
-  hostsTeamId: number | null;
+  hostId: number;
+  hostTeamId: number | null;
+  router: InjectedRouter; // v3
+  currentTeamId: number | undefined;
 }
 
 const baseClass = "select-query-modal";
 
 const SelectQueryModal = ({
   onCancel,
-  onQueryHostCustom,
-  onQueryHostSaved,
-  queries,
-  queryErrors,
   isOnlyObserver,
-  hostsTeamId,
+  hostId,
+  hostTeamId,
+  router,
+  currentTeamId,
 }: ISelectQueryModalProps): JSX.Element => {
+  const { setSelectedQueryTargetsByType } = useContext(QueryContext);
+
+  const { data: queries, error: queriesErr } = useQuery<
+    IListQueriesResponse,
+    Error,
+    ISchedulableQuery[],
+    IQueryKeyQueriesLoadAll[]
+  >(
+    [
+      {
+        scope: "queries",
+        teamId: hostTeamId || API_ALL_TEAMS_ID,
+        mergeInherited: hostTeamId !== API_ALL_TEAMS_ID,
```

Comment:
@sgress454 most of this diff is from moving the API call and associated logic down from the parent level, which never used the response, to the modal level as an optimization not to call for queries every time the user loads the host details page.

Adding lines 61 and 62 are the actual solution to this bug. They tell the server to a) return the team-level queries, whether the team is a "team" or no team (global), and b) if the team is _not_ global, include global queries ("merge inherited") in the response as well.

Reply:
Makes sense, thanks for the context!